UnityProject/Assets/Scripts/HealthV2/Living/PolymorphicSystems/BatterySystem.cs

```diff
@@ -1,17 +1,169 @@
 using System.Collections.Generic;
+using System.Linq;
+using HealthV2.Living.Cyborg;
 using HealthV2.Living.PolymorphicSystems.Bodypart;
+using Systems.Construction.Parts;
 using UnityEngine;
 
 namespace HealthV2.Living.PolymorphicSystems
 {
-	public class BatterySystem : HealthSystemBase
+	public class BatterySystem : HealthSystemBase, IChargeable
 	{
 		public List<BatteryPack> BatteryPacks = new List<BatteryPack>();
 
 		public List<BatteryPoweredComponent> Consuming = new List<BatteryPoweredComponent>();
 
 		public float ConsumingWatts = 0;
 
+
+		public float PreviousStoredWatts;
+
+		public BatteryState CashedBatteryState = BatteryState.FullyCharged;
+
+		private BodyAlertManager BodyAlertManager;
+
+		public override void InIt()
+		{
+			base.InIt();
+			BodyAlertManager = Base.GetComponent<BodyAlertManager>();
+		}
+
+
+		public bool IsFullyCharged
+		{
+			get
+			{
+				bool fullyCharged = true;
+				foreach (var batteryPack in BatteryPacks)
+				{
+					if (batteryPack.IsFullyCharged == false)
+					{
+						fullyCharged = false;
+						break;
+					}
+				}
+
+				return fullyCharged;
+			}
+		}
+
+
+
+		public void ChargeBy(float watts)
```

Comment:
since this is identical in all instances I can read, why not make it the default implementation of the interface?

Reply:
if it gets implemented by  the class I don't think you can have a default implementation, you can have a pre-implemented function that calls it tho, and also requires passing through the battery script into the interface